### DISCUSSION TITLE
Correct way to set up jupyter lab with pdm and PEP 582

### DISCUSSION THREAD

**Discussion by Hofsmo** [pdm-project/pdm#1839] [Q&A] [answered]
Hi,
I had some trouble setting up jupyterlab with pdm and PEP 582. The trouble was encountered with ubunut from WSL (Windows Subsystem for Linux). In the end I managed to make it work nicely, but I feel that I should have been able to find an easier way. Any help or input would be appreciated.

I enabled PEP 582 globally and made the .env file as explained in https://pdm.fming.dev/latest/usage/pep582/ . I installed jupyter lab with the command `pdm add -dG examples jupyterlab`. 

When I try to run `jupyterlab` in the shell I get `jupyterlab: command not found`. The command `python3 -m jupyterlab` does also not work and gives me the following error `JupyterLab application assets not found in "/usr/share/jupyter/lab"`.  

I am able to run it correctly with the command `python3 -m jupyterlab --app-dir /<path_to_app_dir>/__pypackages__/3.10/share/jupyter/lab`. However, I am wondering if there is a simpler way to fix this? I guess I could create a config file for each project that tells jupyter where the app dir is, but that does not seem simpler. Does anyone have simpler solutions?

**Comment by AdamJel** [comment, marked as answer]
@Hofsmo As a side note, PEP-582 has been [rejected](https://discuss.python.org/t/pep-582-python-local-packages-directory/963/429) by Python Steering Council, so it become advisable not to use it.

[PEP-582 status](https://peps.python.org/pep-0582/)

**Reply by Hofsmo** [reply]
I see, I was looking forward to not use virtual environments again :( . I will go back to using virtual environments then, thanks for the reply :) .

**Reply by AdamJel** [reply]
I am also disappointed, but what can you do. Well, the one thing you can do is mark my answer as accepted, if you don't plan to pursue the 582 road :)
